openlcs/libs/unpack.py

```diff
@@ -75,21 +94,18 @@ def unpack_file(file_name, main_dir=None):
         abs_path = os.path.abspath(file_name)
         file_name = os.path.basename(file_name)
         file_path = main_dir or os.path.dirname(abs_path)
-        cmd = ("file -i -b -L '%s' | grep -q 'text'" % file_name)
-        retcode, _ = run(cmd, workdir=file_path, can_fail=True)
-        if retcode == 0:
-            return None
-
-        tmp_dir = tempfile.mkdtemp(prefix='tmpunpack_', dir=main_dir)
-        cmd = ("atool -X '%(tmp_dir)s' -q '%(file_path)s/%(file_name)s' "
-               ">/dev/null" % locals())
-        try:
-            run(cmd, workdir=file_path)
-            return tmp_dir
-        except RuntimeError:
-            shutil.rmtree(tmp_dir)
-            err_msg = 'Unable to decompress file %s.' % file_name
-            raise ValueError(err_msg) from None
+        _, file_extension = get_extension(file_name, SP_EXTENSIONS)
+        if file_extension in SUPPORTED_FILE_EXTENSIONS:
+            tmp_dir = tempfile.mkdtemp(prefix='tmpunpack_', dir=main_dir)
+            cmd = ("atool -X '%(tmp_dir)s' -q '%(file_path)s/%(file_name)s' "
```

Comment:
Here we use `%(file_path)s/%(file_name)s`, but `file_path = main_dir or os.path.dirname(abs_path)`, so if `file_path` use `main_dir `, and `main_dir ` does not have `file_name`, it will display error. So I think should update the logic here

Reply:
Such as:
```
    def unpack_file(file_name, main_dir=None):
        abs_path = os.path.abspath(file_name)
        file_path = main_dir or os.path.dirname(abs_path)

        file_extension = ''.join(pathlib.Path(abs_path).suffixes)
        if file_extension in SUPPORTED_FILE_EXTENSIONS:
            tmp_dir = tempfile.mkdtemp(prefix='tmpunpack_', dir=main_dir)
            cmd = ("atool -X '%(tmp_dir)s' -q '%(abs_path)s' >/dev/null" % locals())
            try:
                run(cmd, workdir=file_path)
                return tmp_dir
            except RuntimeError:
                shutil.rmtree(tmp_dir)
                err_msg = 'Unable to decompress file %s.' % file_name
                raise ValueError(err_msg) from None
```
I don't test the code, but I think it is better to optimize the code